content/blog/2019/root-transition/index.md

```diff
@@ -0,0 +1,142 @@
+---
+title: User guide for extenting Istio root certificate lifetime
+description: The user guide to extend Istio root certificate lifetime.
+publishdate: 2019-06-06
+subtitle:
+attribution: Oliver Liu
+twitter:
+keywords: [security, PKI, certificate, Citadel]
+
+---
+
+## Background
+
+For Istio releases prior to 1.0.8 and prior to 1.1.8, the Istio self-signed certificates have the default lifetime of 1 year.
+If you are using the Istio self-signed certificates with default lifetime configuration,
+please schedule a root transition before it expires. An expiration of root certificate may lead to an unexpected cluster-wide outage.
+After the transition, the root certificate will be renewed to have 10 year lifetime.
+As a best practice for security, we strongly recommend you to do annual root key/cert rotation.
+We will send out instructions for root key/cert rotation as a follow-up.
+
+To evaluate the lifetime remaining for your root certificate, please refer to Step 0 in the following procedure.
+
+We provide the following procedure for you to do the root transition.
+Note that the Envoy instances will be hot restarted to reload the new root certificates, which may impact the long-lived connections.
+For details about the impacts and how Envoy hot restart works, please refer to
+[here](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/hot_restart#arch-overview-hot-restart) and
+[here](https://blog.envoyproxy.io/envoy-hot-restart-1d16b14555b5).
+
+## Scenarios
+
+If you are not currently using the mTLS feature in Istio and will not use it in the future,
+you are not affected and should only upgrade to 1.1.8 or later versions.
+
+If you are not currently using the mTLS feature in Istio and may use it in the future,
+you are recommended to follow the procedure listed below to upgrade.
+
+If you are currently using the mTLS feature in Istio with self-signed certificates,
+please follow the procedure and check whether you will be affected.
+
+## Procedure
+
+### Step 0. Check when your root certificate will expire
+
+Download this [script](TODO) on a machine that has kubectl access to the cluster.
```

Comment:
fix TODO?

Reply:
use backticks around commands (such as kubectl) that are not in bash segments.

Reply:
@wenchenglu @duderino where do you think we should host this script?

Reply:
could we check this script into somewhere in istio github repo? alternatively, we can put it under GCS and make it public accessible.